### PR TITLE
Adds the cyborg screen head

### PR DESCRIPTION
**Pull request by BatElite**
[feat]
## About the PR 
This PR adds the cyborg screen head, whose main feature is that its owner can change the expression on display much like AIs and ghost drones can. You also look like a doofus with one of these.

Under the hood, the functionality is a special case of the eye colour overlay, so you can change the colour easily in a docking station. Changing the expression is done by a new command in the Robot Commands tab. (I couldn't find a nicer way to do it with current borg HUD code) There's 11 expressions available, in either light-on-dark (lod) or dark-on-light mode.

Stat-wise the head has 90 health (vs 50 for light and 175 for standard) and is sorta expensive to make, so it should fit well into the looks vs functionality thing some peeps around here seem to like.

afbeelding

It doesn't work well with hats, but that's the price you pay for fashion.

## Why's this needed? 

It's cool and lets borgs do fun facial expression stuff that AIs and ghostdrones can.

## Changelog

```
(u)BatElite:
(*)Added the cyborg screen head, so that emotionally inclined cyborgs can share their feelings with the world visually! Ask your roboticist to make you one, and check your Robot Commands tab to change expressions.
```

### CODE REVIEW

**Comment by BatElite** [comment]
Bonus screenshot:
afbeelding

**Comment by BatElite** [comment]
I don't really get that error, AFAIK I'm calling update_bodypart as intended

**Comment by Tiggersaurus** [comment]
Oh my gosh yes 😍

**Comment by mordent-goonstation** [comment]
I like the idea, but think the "flat holo-screen" thing is weird. Big CRT screens I can get behind.

**Comment by flrsh** [comment]
I would prefer a hologram that cyborgs could project above their head instead of replacing their head. They look pretty odd in this current implementation.

**Comment by BatElite** [comment]
I'll see if moving the screen in a bit for the side views help WRT silhouette, they poke out rather a lot now on light borgs. IIRC on normal borgs it looks a bit more sensible.
I'm not against turning it into a CRT thing, though despite the size that'd have I'm hesitant about adding a speed penalty more than being a standard head would give. Since most borg players prefer speedy builds, I worry that would tip it over from "gimmicky tradeoff" into "not worth bothering with". The design for it as is was Nanotrasen bolting a panel screen awkwardly onto the smallest case they could fit a brain in. It's also possible both could exist side by side (the CRT as a heavy variant)
(Maybe we should have a talk about making "being a heavy borg" actually worthwhile sometime, but that's supplementary to this)

As for holograms, I'm primarily concerned about difficulty wrt readability on busy backgrounds. There's also that doing it like that would leave no "faceless" borg options, which I can see having an aesthetic appeal.

**Comment by Gannets** [comment]
I am all about crt head

**Comment by BatElite** [comment]
Haven't gotten feedback on these yet in imspriter (US nighttime lull I'm guessing), but here's where we're at:

afbeelding

**Comment by Emcee-Gore** [comment]
> 
> 
> Haven't gotten feedback on these yet in imspriter (US nighttime lull I'm guessing), but here's where we're at:
> 
> afbeelding
Damn I love this idea! I think the side-views and back-view is great.
The front view looks a bit strange like there is no border for the screen?

**Comment by BatElite** [comment]
I did try a border but the result was pretty comically huge since the actual screen hasn't changed (changing all the faces is tedium and I think the shape is nice as is). Here's what that looked like (the sides were also taller):
 
afbeelding

 
 Also here's the latest push (and thus what's actually in the PR):

afbeelding


On a general note, I think whatever sprite changes might happen are going to be minor, so I'd like for someone to move along and look at whether the code's good. It's nearing on a month.

File: code/mob/living/silicon/robot.dm [hunk 1]
```diff
@@ -2101,6 +2101,26 @@
...
 					boutput(src, "<span class='alert'>You have locked your interface.</span>")
 
+	verb/cmd_alter_head_screen()
+		set category = "Robot Commands"
+		set name = "Change facial expression (screen head only)"
+		var/obj/item/parts/robot_parts/head/screen/targethead = locate(/obj/item/parts/robot_parts/head/screen) in src.contents
+		if (!istype(targethead))
+			boutput(src, "<span class='alert'>You're not equipped with a suitable head to use this command!</span>")
+			return 0
+		//for (var/obj/item/parts/robot_parts/head/screen/ in src.contents)
```
Comment (pali6) [inline code/mob/living/silicon/robot.dm:2111]:
```suggestion
```

File: code/mob/living/silicon/robot.dm [hunk 2]
```diff
@@ -2101,6 +2101,26 @@
...
+
+		var/newFace = input(usr, "Select your faceplate", "Face settings", targethead.face) as null|anything in targethead.expressions
+		if (!newFace) return 0
+		var/newMode = input(usr, "Select a display mode", "Face settings", targethead.mode) as null|anything in list("lod", "dol")
```
Comment (pali6) [inline code/mob/living/silicon/robot.dm:2115]: Maybe something more descriptive than lod and dol?

**Review by pali6 — APPROVED** [review]
I like it, nice and cute.

File: code/mob/living/silicon/robot.dm [hunk 3]
```diff
@@ -2108,13 +2108,13 @@
...
 			return 0
-		//for (var/obj/item/parts/robot_parts/head/screen/ in src.contents)
 
 		var/newFace = input(usr, "Select your faceplate", "Face settings", targethead.face) as null|anything in targethead.expressions
 		if (!newFace) return 0
-		var/newMode = input(usr, "Select a display mode", "Face settings", targethead.mode) as null|anything in list("lod", "dol")
+		var/newMode = input(usr, "Select a display mode", "Face settings", targethead.mode) as null|anything in list("light-on-dark", "dark-on-light")
 		if (!newMode) return 0
 		newFace = (newFace ? lowertext(newFace) : targethead.face)
+		newMode = (newMode == "light-on-dark" ? "lod" : "dol")
```
Comment (BatElite) [inline code/mob/living/silicon/robot.dm:2117]: This is probably dirty but I couldn't get an associative list to work in the input.

Comment (pali6) [reply code/mob/living/silicon/robot.dm:2117]: Eh, it's fine.

**Comment by pali6** [comment]
I've played around a bit with the sprite to give it a frame without making it wider:
dreamseeker_clSkLM1OQX
dreamseeker_AyrH6mTxmc
dreamseeker_G3aQGYmvK7
dreamseeker_k5Tl1WLMB7

How does that look?

**Comment by mordent-goonstation** [comment]
> I've played around a bit with the sprite to give it a frame without making it wider:
> dreamseeker_clSkLM1OQX
> dreamseeker_AyrH6mTxmc
> dreamseeker_G3aQGYmvK7
> dreamseeker_k5Tl1WLMB7
> 
> How does that look?

This is my personally favorite variant. I'm not against BatElite's one ([here](https://github.com/goonstation/goonstation/pull/3022#issuecomment-755322595)) either.

**Comment by BatElite** [comment]
A murder??

afbeelding

They're a bit rounder now but bordered. Like borgstronauts.

**Pull request merged**